omnigan/trainer.py

```diff
@@ -221,13 +220,35 @@ def setup(self):
         self.G: OmniGenerator = get_gen(self.opts, verbose=self.verbose).to(self.device)
         print("Generator OK. Computing latent & input shapes...", end="", flush=True)
 
+        if self.G.encoder is not None:
+            self.latent_shape = self.compute_latent_shape()
+
+
         self.input_shape = self.compute_input_shape()
         if "s" in self.opts.tasks:
             self.G.decoders["s"].set_target_size(self.input_shape[-2:])
 
         print("OK.")
         self.painter_z_h = self.input_shape[-2] // (2 ** self.opts.gen.p.spade_n_up)
         self.painter_z_w = self.input_shape[-1] // (2 ** self.opts.gen.p.spade_n_up)
+        if "m2" in self.opts.tasks:
+            self.label_1 = torch.zeros(
```

Comment:
maybe `label_0` and `label_1` or `label_low` `label_high` would make more sense

Reply:
can we actuallly put low high ?

Reply:
I kept it as numbers so that we can add more flood levels as needed. But I agree the lowest level should be label 0. I was anticipating bringing in the ground as label 0 (coming soon in an update)

Reply:
oh ok then makes sense, I'm fine with it